### PR TITLE
Feat/add ExamButton component

### LINKED ISSUE #31: [Feature] Page selection button component
Is your feature request related to a problem? Please describe.
Currently, the page selection buttons from the home page are hardcoded.

Describe the solution you'd like
A new page button component added to the library that uses Tailwind variants for customization.
image

Additional context
[Class Variance Authority](https://github.com/joe-bell/cva) can be used to create the tailwind variants.
Check [button component](../blob/main//components/Button.tsx)

### PR DESCRIPTION
**Pull request by Lucif3r-in**
Fixes #31 

- [x] I have read and agree to the [Code of Conduct](https://github.com/eduardconstantin/azure-fundamentals/blob/main/CODE_OF_CONDUCT.md), and have followed the [Contributing guidelines](https://github.com/eduardconstantin/azure-fundamentals/blob/main/CONTRIBUTING.md).

## Pull Request Type

- [x] New feature

## Summary
Added ExamButton component using CVA

### CODE REVIEW

**Comment by Lucif3r-in** [comment]
@eduardconstantin Please review this!

File: components/ExamButton.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+export interface LinkProps
+  extends React.AnchorHTMLAttributes<HTMLAnchorElement>,
+    VariantProps<typeof link> {
+  href: string;
+  heading: string;
+  paragraph: string;
+  containerClassName?: string;
+  headingClassName?: string;
+  paragraphClassName?: string;
```
Comment (eduardconstantin) [inline components/ExamButton.tsx:53]:
```suggestion

```

File: components/ExamButton.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        "cursor-pointer",
+      ],
+      exam: [
+        "bg-slate-700",
+        "hover:bg-gradient-to-r",
+        "from-[#F97316]",
+        "to-[#FACC15]",
+        "rounded-xl",
+        "p-[2px]",
+        "w-[350px]",
```
Comment (eduardconstantin) [inline components/ExamButton.tsx:26]:
```suggestion

```

File: components/ExamButton.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        "h-[350px]",
```
Comment (eduardconstantin) [inline components/ExamButton.tsx:27]:
```suggestion

```

File: components/ExamButton.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  variants: {
+    mode: {
+      practice: [
+        "bg-slate-700",
+        "hover:bg-gradient-to-r",
+        "from-[#0284C7]",
+        "to-[#2DD48F]",
+        "rounded-xl",
+        "p-[2px]",
+        "w-[350px]",
```
Comment (eduardconstantin) [inline components/ExamButton.tsx:15]:
```suggestion

```

File: components/ExamButton.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        "h-[350px]",
```
Comment (eduardconstantin) [inline components/ExamButton.tsx:16]:
```suggestion

```

File: components/ExamButton.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        "cursor-pointer",
+      ],
+      exam: [
+        "bg-slate-700",
+        "hover:bg-gradient-to-r",
+        "from-[#F97316]",
+        "to-[#FACC15]",
+        "rounded-xl",
```
Comment (eduardconstantin) [inline components/ExamButton.tsx:24]:
```suggestion

```

File: components/ExamButton.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+const link = cva("a", {
+  variants: {
+    mode: {
+      practice: [
+        "bg-slate-700",
+        "hover:bg-gradient-to-r",
+        "from-[#0284C7]",
+        "to-[#2DD48F]",
+        "rounded-xl",
```
Comment (eduardconstantin) [inline components/ExamButton.tsx:13]:
```suggestion

```

File: components/ExamButton.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        "h-[350px]",
+        "cursor-pointer",
+      ],
+      exam: [
+        "bg-slate-700",
+        "hover:bg-gradient-to-r",
+        "from-[#F97316]",
+        "to-[#FACC15]",
+        "rounded-xl",
+        "p-[2px]",
```
Comment (eduardconstantin) [inline components/ExamButton.tsx:25]:
```suggestion

```

File: components/ExamButton.tsx [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+const link = cva("a", {
+  variants: {
+    mode: {
+      practice: [
+        "bg-slate-700",
+        "hover:bg-gradient-to-r",
+        "from-[#0284C7]",
+        "to-[#2DD48F]",
+        "rounded-xl",
+        "p-[2px]",
```
Comment (eduardconstantin) [inline components/ExamButton.tsx:14]:
```suggestion

```

**Review by eduardconstantin — CHANGES_REQUESTED** [review]

File: pages/index.tsx [hunk 10]
```diff
@@ -1,42 +1,37 @@
...
-            <p className="text-sm text-slate-400 mt-7">
-              Learn and familiarize yourself with the questions and answers without any time constraint.
-            </p>
-          </div>
-        </Link>
-        <Link
+          mode="practice"
+          heading="Practice mode"
+          paragraph="Learn and familiarize yourself with the questions and answers without any time constraint."
+          containerClassName="flex flex-col h-full group bg-slate-800 rounded-xl pt-[30%] px-7"
```
[the remark this answers is not in the document]
Comment (Lucif3r-in) [reply pages/index.tsx:21]: @eduardconstantin So we we only need to have CustomLink right? Data inside should remain as it is?

[the remark this answers is not in the document]
Comment (eduardconstantin) [reply pages/index.tsx:21]: no, remove only the containerClassName, headingClassName and paragraphClassName props, these are not changing so you can hardcode them inside the component

**Pull request closed without merging**